modules/sharedIdSystem.js

```diff
@@ -1,373 +1,107 @@
-    return () => {
-      const buffer = new Uint8Array(1);
-      browserCrypto.getRandomValues(buffer);
-      return buffer[0] / 0xff;
-    };
-  }
-  return () => Math.random();
-}
-
-/**
- * Builds and returns the shared Id URL with attached consent data if applicable
- * @param {Object} consentData
- * @return {string}
- */
-function sharedIdUrl(consentData) {
-  const usPrivacyString = uspDataHandler.getConsentData();
-  let sharedIdUrl = ID_SVC;
-  if (usPrivacyString) {
-    sharedIdUrl = `${ID_SVC}?us_privacy=${usPrivacyString}`;
-  }
-  if (!consentData || typeof consentData.gdprApplies !== 'boolean' || !consentData.gdprApplies) return sharedIdUrl;
-  if (usPrivacyString) {
-    sharedIdUrl = `${sharedIdUrl}&gdpr=1&gdpr_consent=${consentData.consentString}`
-    return sharedIdUrl;
-  }
-  sharedIdUrl = `${ID_SVC}?gdpr=1&gdpr_consent=${consentData.consentString}`;
-  return sharedIdUrl
-}
+const GVLID = 887;
 
 /** @type {Submodule} */
-export const sharedIdSubmodule = {
+export const pubCommonIdSubmodule = {
   /**
    * used to link submodule with config
    * @type {string}
    */
-  name: MODULE_NAME,
-
+  name: 'sharedId',
+  aliasName: 'pubCommonId',
   /**
-   * Vendor id of Prebid
+   * Vendor id of prebid
    * @type {Number}
    */
   gvlid: GVLID,
+  /**
+   * Return a callback function that calls the pixelUrl with id as a query parameter
+   * @param pixelUrl
+   * @param id
+   * @returns {function}
+   */
+  makeCallback: function (pixelUrl, id = '') {
+    if (!pixelUrl) {
+      return;
+    }
+
+    // Use pubcid as a cache buster
+    const urlInfo = utils.parseUrl(pixelUrl);
+    urlInfo.search.id = encodeURIComponent('pubcid:' + id);
+    const targetUrl = utils.buildUrl(urlInfo);
+
+    return function () {
+      utils.triggerPixel(targetUrl);
+    };
+  },
   /**
    * decode the stored id value for passing to bid requests
    * @function
    * @param {string} value
-   * @returns {{sharedid:{ id: string, third:string}} or undefined if value doesn't exists
+   * @param {SubmoduleConfig} config
+   * @returns {{pubcid:string}}
    */
-  decode(value) {
-    return (value) ? encodeId(value) : undefined;
+  decode(value, config) {
+    return {'pubcid': value};
   },
-
   /**
-   * performs action to obtain id and return a value.
+   * performs action to obtain id
    * @function
-   * @param {SubmoduleConfig} [config]
-   * @param {ConsentData|undefined} consentData
-   * @returns {sharedId}
+   * @param {SubmoduleConfig} [config] Config object with params and storage properties
+   * @param {Object} consentData
+   * @param {string} storedId Existing pubcommon id
+   * @returns {IdResponse}
    */
-  getId(config, consentData) {
-    const coppa = coppaDataHandler.getCoppa();
-    if (coppa) {
-      utils.logInfo('SharedId: IDs not provided for coppa requests, exiting SharedId');
-      return;
+  getId: function (config = {}, consentData, storedId) {
+    const {params: {create = true, pixelUrl} = {}} = config;
```

Comment:
Is the coppa condition no longer needed? It seems to have been dropped

Reply:
As I understand coppa is related to sharedId = and I need to remove all refernces to sharedId in this file, so I decided to remove coppa too

Reply:
updated